akka-docs-dev/rst/scala/code/docs/stream/StreamPartialFlowGraphDocSpec.scala

```diff
@@ -142,4 +144,62 @@ class StreamPartialFlowGraphDocSpec extends AkkaSpec {
 
     Await.result(matSink, 300.millis) should equal(1 -> "1")
   }
+
+  "help users in providing complex ports" in {
+    object AudioPorts extends Ports {
+      val nums = Input[Int] // TODO: use I / O words or keep sinks/sources?
+      val words = Input[String]
+
+      val out = Output[String]
+    }
+
+    // building
+    val g = PartialFlowGraph(AudioPorts) { implicit b ⇒
+      import FlowGraphImplicits._
+      val merge = Merge[String]
+      // format: OFF
+      AudioPorts.nums ~> Flow[Int].map(_.toString) ~> merge
+                                  AudioPorts.words ~> merge ~> AudioPorts.out
+      // format: ON
+    }
+
+    // connecting, current style:
+    FlowGraph(g) { b ⇒
```

Comment:
It is possible to have `AudioPorts` captured in `g` and have some instance class with all actual `Source`s and `Sink`s captured and bind not every in and out, but `g` and actual instance of `AudioPorts`. 

Right now I could not provide API, but just an idea how to improve type safety and force to have all in/outs get bound.